### DISCUSSION TITLE
Alternatives for StrongSwan iOS client.

### DISCUSSION THREAD

**Discussion by Jdgmx** [strongswan/strongswan#1834] [Q&A]
Hello,

We are developing a VPN client for iOS that connects to a Strong Swan server using certificates for authenticating and authorization. For that we need to use a root CA certificate that is self signed. 
We have no issues on the Strong Swan side since we have control to manually trust the root certificate
But on the iOS side, since we use the system IKEv2 client, there is no way to change the behavior of the system trust evaluator. Meaning that there is no way to programmatically trust our custom root certificate.

The alternatives that we have are, 1st, use a mobile profile to download the certificate, but this solution is too cumbersome for our users. 2nd, implement our own version of the IKEv2 protocol totally within our application using the NetworkExtension framework and the Packet Tunnel Provider API.

We have been looking for alternatives to implement the 2nd solution but haven't found anything suitable. 
Our questions are, is there any framework that you know that implements IKEv2 using the the Packet Tunnel Provider API?. Or are there any other alternatives to solve the issue?

Thank you.

**Comment by ptchuba** [comment]
I'm interested in this as well. Any suggestions??

**Comment by tobiasbrunner** [comment]
See my response here: https://github.com/strongswan/strongswan/discussions/1789#discussioncomment-6424479
